### PR TITLE
Fix reversible args updates bug

### PR DESCRIPTION
**Pull request by ilya16**
Filter object with `SettableArgs` modules was iterated only during the first `model.forward()` call and then became empty. As a result, `**kwargs` with input and context masks were ignored on each subsequent `model.forward()` call.

Caught this error while training the Reformer model with `input_mask` and varying `batch_size`.

Bug was introduced in https://github.com/lucidrains/reformer-pytorch/commit/d2483dc88478f686a40cccb83b1759cbc338defa.

### CODE REVIEW

**Comment by lucidrains** [comment]
@ilya16 oh gosh, thank you for catching this! are you training an encoder / decoder architecture?

**Comment by lucidrains** [comment]
@ilya16 oh wow, you go to Skoltech! I visited there for the Skolkovo AI conference some time ago! Very nice school

**Comment by ilya16** [comment]
@lucidrains hi, thanks! It's nice to find that you know Skoltech :)

For now I'm experimenting with a single Reformer model as part of my research project, the input and target sequences are from the same domain.

By the way, thanks for this Reformer implementation. Learned new things in Pytorch and understood Reformer architecture better while digging into it.

**Comment by lucidrains** [comment]
@ilya16 no problem! thanks for catching the bug and have a great day!
